src/H5Eint.c

```diff
@@ -98,6 +120,738 @@ char H5E_mpi_error_str[MPI_MAX_ERROR_STRING];
 int  H5E_mpi_error_str_len;
 #endif /* H5_HAVE_PARALLEL */
 
+/* First & last major and minor error codes registered by the library */
+hid_t H5E_first_maj_id_g = H5I_INVALID_HID;
+hid_t H5E_last_maj_id_g  = H5I_INVALID_HID;
+hid_t H5E_first_min_id_g = H5I_INVALID_HID;
+hid_t H5E_last_min_id_g  = H5I_INVALID_HID;
+
+/* Error class ID class */
+static const H5I_class_t H5I_ERRCLS_CLS[1] = {{
+    H5I_ERROR_CLASS,                  /* ID class value */
+    0,                                /* Class flags */
+    0,                                /* # of reserved IDs for class */
+    (H5I_free_t)H5E__unregister_class /* Callback routine for closing objects of this class */
+}};
+
+/* Error message ID class */
+static const H5I_class_t H5I_ERRMSG_CLS[1] = {{
+    H5I_ERROR_MSG,             /* ID class value */
+    0,                         /* Class flags */
+    0,                         /* # of reserved IDs for class */
+    (H5I_free_t)H5E__close_msg /* Callback routine for closing objects of this class */
+}};
+
+/* Error stack ID class */
+static const H5I_class_t H5I_ERRSTK_CLS[1] = {{
+    H5I_ERROR_STACK,             /* ID class value */
+    0,                           /* Class flags */
+    0,                           /* # of reserved IDs for class */
+    (H5I_free_t)H5E__close_stack /* Callback routine for closing objects of this class */
+}};
+
+/* Library's error class */
+static const H5E_cls_t H5E_err_cls_s = {false, H5E_CLS_NAME, H5E_CLS_LIB_NAME, H5_VERS_STR};
+
```

Comment:
Autogenerated headers that define the major and minor error messages for the library.  Used to register the library's error code IDs.